src/middlewared/middlewared/etc_files/localtime_config.py

```diff
@@ -1,19 +1,36 @@
+import contextlib
 import os
 import shutil
+import subprocess
+
+from middlewared.utils import osc
 
 
 def localtime_configuration(middleware):
     system_config = middleware.call_sync('system.general.config')
     if not system_config['timezone']:
         system_config['timezone'] = 'America/Los_Angeles'
 
-    shutil.copy(
-        os.path.join('/usr/share/zoneinfo/', system_config['timezone']),
-        '/etc/localtime'
-    )
+    if osc.IS_LINUX:
+        with contextlib.suppress(OSError):
+            os.unlink('/etc/localtime')
+        os.symlink(os.path.join('/usr/share/zoneinfo', system_config['timezone']), '/etc/localtime')
+        cp = subprocess.Popen(
+            ['systemctl', 'daemon-reload'], stdout=subprocess.DEVNULL, stderr=subprocess.PIPE
+        )
+        _, stderr = cp.communicate()
+        if cp.returncode:
+            middleware.logger.error(
+                'Failed to reload systemctl daemon after timezone configuration: %s', stderr.decode()
```

Comment:
Verbiage seems strange here. Maybe you mean `Failed to reload ntp daemon`?

Reply:
Nope, we are not reloading ntp daemon. The intention is to reload systemctl daemon so it reflects the timezone change as well.